src/lib.rs

```diff
@@ -1201,6 +1214,61 @@ impl<'a, T> EventListenerFuture for RecvInner<'a, T> {
     }
 }
 
+easy_wrapper! {
+    /// A future returned by [`Receiver::recv_owned()`].
+    #[derive(Debug)]
+    #[must_use = "futures do nothing unless you `.await` or poll them"]
+    pub struct RecvOwned<T>(RecvOwnedInner<T> => Result<T, RecvError>);
+    #[cfg(all(feature = "std", not(target_family = "wasm")))]
+    pub(crate) wait();
+}
+
+pin_project! {
+    #[derive(Debug)]
+    #[project(!Unpin)]
+    struct RecvOwnedInner<T> {
+        // Reference to the receiver.
+        receiver: Receiver<T>,
+
+        // Listener waiting on the channel.
+        listener: Option<EventListener>,
+
+        // Keeping this type `!Unpin` enables future optimizations.
+        #[pin]
+        _pin: PhantomPinned
+    }
+}
+
+impl<T> EventListenerFuture for RecvOwnedInner<T> {
+    type Output = Result<T, RecvError>;
+
+    /// Run this future with the given `Strategy`.
+    fn poll_with_strategy<'x, S: Strategy<'x>>(
+        self: Pin<&mut Self>,
+        strategy: &mut S,
+        cx: &mut S::Context,
+    ) -> Poll<Result<T, RecvError>> {
+        let this = self.project();
+
+        loop {
+            // Attempt to receive a message.
+            match this.receiver.try_recv() {
+                Ok(msg) => return Poll::Ready(Ok(msg)),
+                Err(TryRecvError::Closed) => return Poll::Ready(Err(RecvError)),
+                Err(TryRecvError::Empty) => {}
+            }
+
+            // Receiving failed - now start listening for notifications or wait for one.
+            if this.listener.is_some() {
+                // Poll using the given strategy
+                ready!(S::poll(strategy, &mut *this.listener, cx));
+            } else {
+                *this.listener = Some(this.receiver.channel.recv_ops.listen());
+            }
+        }
+    }
+}
+
```

Comment:
Rather than duplicating all of this code from `Recv`, why not extract it into a common type? For instance, you could do this:

```rust
struct RecvInner<T, Channel: Borrow<Receiver<T>> { ... }
```

...then do this:

```rust
pub struct Recv<'a, T>(RecvInner<T, &'a Receiver<T>>);
pub struct RecvOwned<T>(RecvInner<T, Receiver<T>>);
```

This would prevent duplicating around 250 lines of code.

Reply:
I'm playing around with this suggestion, but the compiler complains about `T` being unused.

I'll figure it out.

Reply:
Use [`PhantomData`](https://doc.rust-lang.org/std/marker/struct.PhantomData.html) to ensure that `T` is captured

Reply:
I propose 01fbd7dc698748e3f612525e414bffd6f21824d8